src/open_inwoner/search/tests/test_page.py

```diff
@@ -312,7 +312,6 @@ def _click_checkbox_for_name(page=page, name=''):
 
             # Open het <details>-element
             # details_element.click()
```

Comment:
You need to query the `detail_element` with `get_by_text` and pass in the `name` to get the correct locator:

```
details_element = page.locator(".filter").filter(has=page.get_by_text(name))
details_element.click()
```

This will make the relevant `details_element` (a different one for each call to `_click_checkbox_for_name` visible